### PR TITLE
Connect api lambda to db

### PR DESCRIPTION
**Pull request by MDunitz**
# Need
- as a user I need to submit a query via the api and get results from the db

# Approach
- /query endpoint accepts a query string in the body of the request
- the api lambda submits a read only query to the db and returns the response and the original query to the user

# Test
- manually tested read only permissions
- unit tests for the endpoint
- manually tested functionality
Can test by going to https://query.dev.data.humancellatlas.org/ and trying the /query endpoint with a query such as `SELECT * FROM files LIMIT 10;`

# ToDo
- include column names in query results
- better error handling around timeouts (`SELECT * FROM files;` returns a 504 with no error message) 
# Questions
- how to handle write attempts? Currently returning a 500

### CODE REVIEW

File: config/query-api.yml [hunk 1]
```diff
@@ -35,6 +35,32 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 
+  /v1/query:
+    post:
+      summary: Log query
+      description: Print Query
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply config/query-api.yml]: 👍

File: config/iam-policy-templates/query-api-lambda.json [hunk 2]
```diff
@@ -20,8 +20,18 @@
...
+
       ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "xray:PutTraceSegments",
+        "xray:PutTelemetryRecords"
+      ],
+      "Resource": "*"
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply config/iam-policy-templates/query-api-lambda.json]: 👍

File: query/lib/config.py [hunk 3]
```diff
@@ -33,7 +35,7 @@ def __setattr__(self, attr, value):
     _secret = json.loads(AwsSecret(f"dcp/query/{deployment_stage}/database").value)
 
     _pg_bouncer_dns_name = _get(_secret, 'pgbouncer_dns_name')
-    _rds_dns_name = _get(_secret, 'rds_dns_name')
+    # _rds_dns_name = _get(_secret, 'rds_dns_name')
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply query/lib/config.py]: 👍

File: query/requirements.txt [hunk 4]
```diff
@@ -6,3 +6,5 @@ hca==4.4.9
 pylru
 PyYAML==3.13
 connexion==1.5.2
+
+
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply query/requirements.txt]: 👍

File: query/lib/db/bundles.py [hunk 5]
```diff
@@ -3,9 +3,9 @@
 from uuid import UUID
 from psycopg2.extras import Json
 
-from lib.model import datetime_to_version, Bundle
-from lib.config import requires_admin_mode
-from lib.db.table import Table
+from ..model import datetime_to_version, Bundle
+from ..config import requires_admin_mode
+from ..db.table import Table
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply query/lib/db/bundles.py]: 👍

File: query/lib/config.py [hunk 6]
```diff
@@ -1,15 +1,17 @@
 import os
 import json
-from lib.logger import logger
 from dcplib.aws_secret import AwsSecret
 
+from .logger import logger
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply query/lib/config.py]: removed

File: query/scripts/load_db [hunk 7]
```diff
@@ -8,12 +8,13 @@ from multiprocessing import Pool
 pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
 sys.path.insert(0, pkg_root)  # noqa
 
-from lib.config import Config
-from lib.db.database import PostgresDatabase
-from lib.etl.load import PostgresLoader, Loader
-from lib.logger import logger
-from lib.model import datetime_to_version
-from lib.etl.extract import S3Client, S3Extractor, Extractor
+
```
Comment (mweiden) [inline query/scripts/load_db]: format: extra newline here

File: query/test/unit/api_server/test_endpoints.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class TestEndpoints(unittest.TestCase):
+
+    def test_heathcheck_endpoint(self):
+        self.client = client_for_test_api_server()
+
+        response = self.client.get(f"/v1/health")
+
+        self.assertEqual(response.status_code, 200)
```
Comment (mweiden) [inline query/test/unit/api_server/test_endpoints.py]: format: spacing is a little different here relative to the other test.

File: query/lambdas/api_server/v1/fast_query.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import json
+
+import requests
+
+from query.lib.common.logging import get_logger
+from query.lib.config import Config
+from query.lib.db.database import PostgresDatabase
+from .. import return_exceptions_as_http_errors
```
Comment (mweiden) [inline query/lambdas/api_server/v1/fast_query.py]: format: one last relative import

File: query/scripts/load_db (hunk 7 above)
Comment (MDunitz) [reply query/scripts/load_db]: 👍

File: query/test/unit/api_server/test_endpoints.py (hunk 8 above)
Comment (MDunitz) [reply query/test/unit/api_server/test_endpoints.py]: 👍

File: query/lambdas/api_server/v1/fast_query.py (hunk 9 above)
Comment (MDunitz) [reply query/lambdas/api_server/v1/fast_query.py]: 👍

**Pull request merged**